producer/data_repository.go

```diff
@@ -36,8 +36,12 @@ const (
 
 var CurrentResourceUri string
 
-func getDataFromDB(collName string, filter bson.M) (map[string]interface{}, *models.ProblemDetails) {
-	data := MongoDBLibrary.RestfulAPIGetOne(collName, filter)
+func getDataFromDB(collName string, filter bson.M, m mongoapi.MongoClient) (map[string]interface{}, *models.ProblemDetails) {
```

Comment:
It appears that this is the only file which is referring to the mongoDB..so instead of passing m to gin callbacks, can we have some method so this module/file gets the updated mongoDB handler? It will greatly reduce the amount of code changes.

Reply:
Hello @thakurajayL Do you have any idea that what kind of methods could be useful here ? Could you please give some examples ?

Reply:
Can you call API from the main and pass mongo Client handler to producer?

Reply:
MongoDBClient is created in producer and only the producer/data_repository.go file is affected.